src/util/migrate.cpp

```diff
@@ -2182,6 +2217,23 @@ typet migrate_type_back(const type2tc &ref)
 
     return std::move(thetype);
   }
+  case type2t::vector_id:
+  {
+    const vector_type2t &ref2 = to_vector_type(ref);
+
+    vector_typet thetype;
+    thetype.subtype() = migrate_type_back(ref2.subtype);
+    if(ref2.size_is_infinite)
+    {
+      thetype.set("size", "infinity");
+    }
```

Comment:
this should never happen, right? Change it to an assertion.

Reply:
Because of
```
        "Vector type has a constant size\n"
        "Please, refer to: "
        "https://clang.llvm.org/docs/"
        "LanguageExtensions.html#vectors-and-extended-vectors"
```